Saule/TypeExtensions.cs

```diff
@@ -33,5 +33,21 @@ public static IEnumerable<Type> GetInheritanceChain(this Type type)
                 .Concat(type.BaseType.GetInheritanceChain())
                 .Distinct();
         }
+
+        public static bool IsEnumerable(this Type type)
+        {
+            return type.IsGenericType && type.GetGenericTypeDefinition() == typeof(IEnumerable<>);
+        }
+
+        /// <summary>
+        /// Get the type of the items if the type is an <see cref="IEnumerable{T}"/> (if not it will return the input type).
+        /// </summary>
+        /// <param name="type">The type to apply this extension method to.</param>
+        /// <returns>The type of the items if the type is an <see cref="IEnumerable{T}"/> (if not it will return the input type).</returns>
+        public static Type TryGetCollectionType(this Type type)
+        {
+            var collectionType = type.GetInterfaces().FirstOrDefault(i => i.IsEnumerable());
+            return collectionType?.GenericTypeArguments[0] ?? type;
```

Comment:
I'd prefer if this just returned `null` if the type could not be found. Now the behavior is really confusing from the name (`TryX` implies returning a `bool` and setting an `out` parameter to me).